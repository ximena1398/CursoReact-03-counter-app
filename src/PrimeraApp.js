//COMPONENTES (Se espera que regrese algo de HTML)
//Functional components
import React from 'react'
import PropTypes from 'prop-types'; //Tipos de las propiedades que recibirá el componente 

const PrimeraApp = ({saludo, subtitulo}) => { //desestructuración de elementos props: en este caso saludo que ya viene de index.js
    //const saludo = "Hola Mundo";

    return (
        <>
            <h1> {saludo} </h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtítulo'
}

export default PrimeraApp;