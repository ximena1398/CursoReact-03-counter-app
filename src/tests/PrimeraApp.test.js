import React from 'react'
//import { render } from "@testing-library/react";
import { shallow } from "enzyme";

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en PrimeraApp', () => {

    // test('debe mostrar el mensaje Hola, Soy Goku ', () => {
    //     const saludo = 'Hola, soy Goku';
    //     //const wrapper = render( <PrimeraApp saludo = {saludo}/> ); //se debe renderizar (se deden enviar por parámetros los datos requeridos)
    //     //wrapper con desestructuración
    //     const { getByText } = render( <PrimeraApp saludo = {saludo}/> );
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('debe mostrar <PrimetaApp/> correctamente ', () => {
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el subtítulo enviado por props ', () => {
        const saludo = 'Hola, Soy Goku';
        const subTitulo = "Soy un subtitulo";

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subTitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text(); //find: buscar en el html el párrafo del subtitulo 'p'
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subTitulo);
    })

})
