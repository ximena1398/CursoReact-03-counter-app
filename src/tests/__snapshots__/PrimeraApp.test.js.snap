// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar <PrimetaApp/> correctamente  1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
